http/src/main/java/io/micronaut/http/util/CookieUtil.java

```diff
@@ -0,0 +1,210 @@
+                    //(multiple semicolons)
+                    if (split[0].trim().equals("")) {
+                        return null;
+                    }
+                    if (split.length < 2) {
+                        return new AbstractMap.SimpleEntry<>(split[0].trim().toLowerCase(), "");
+                    }
+                    return new AbstractMap.SimpleEntry<>(split[0].trim().toLowerCase(), trimQuotes(split[1].trim()));
+                })
+                .filter(Objects::nonNull)
+                .collect(Collectors.toMap(AbstractMap.SimpleEntry::getKey, AbstractMap.SimpleEntry::getValue));
+
+        String cookieName = cookieDefinition[0].substring((HttpHeaders.SET_COOKIE + ":").length()).trim();
+        String cookieValue = trimQuotes(cookieDefinition[1]);
+        return new HeaderPropertyMapCookieConfiguration(cookieName, cookieValue, props);
+    }
+
+    private static String trimQuotes(String propertyValue) {
+        if (propertyValue == null) {
+            return null;
+        }
+        return propertyValue.startsWith("\"") && propertyValue.endsWith("\"")
+                ? propertyValue.substring(1, propertyValue.length() - 1)
+                : propertyValue;
+    }
+
+
+    private static class HeaderPropertyMapCookieConfiguration implements CookieConfiguration {
+
+        private static final DateTimeFormatter ANSI_C_ASCTIME_FORMATTER = new DateTimeFormatterBuilder()
+                .appendPattern("[EEEE][E][,] MMM [ ]")
+                .appendValueReduced(ChronoField.DAY_OF_MONTH, 1, 2, 1)
+                .appendPattern(" HH:mm:ss ")
+                .appendValueReduced(ChronoField.YEAR_OF_ERA, 2, 4, LocalDate.now().minusYears(80))
+                .toFormatter();
+
+        private static final DateTimeFormatter RFC_5322_RFC_850_FORMATTER = new DateTimeFormatterBuilder()
+                .appendPattern("[EEEE][E], dd[-][ ]MMM[-][ ]")
+                .appendValueReduced(ChronoField.YEAR_OF_ERA, 2, 4, LocalDate.now().minusYears(80))
+                .appendPattern(" HH:mm:ss ")
+                .appendZoneOrOffsetId()
+                .toFormatter();
+
+        private final String cookieName;
+        private final String cookieValue;
+        private final Map<String, String> props;
+
+        public HeaderPropertyMapCookieConfiguration(String cookieName, String cookieValue, Map<String, String> props) {
+            this.cookieName = cookieName;
+            this.cookieValue = cookieValue;
+            this.props = props;
+        }
+
+        public Cookie toCookie() {
+            return new SimpleCookie(cookieName, cookieValue)
+                    .configure(this, isCookieSecure().orElse(false));
+        }
+
+        @Override
+        @NonNull
+        public String getCookieName() {
+            return cookieName;
+        }
+
+        @Override
+        public Optional<String> getCookieDomain() {
+            return Optional.ofNullable(props.get(CookieProperties.DOMAIN.toLowerCase()));
```

Comment:
It seems like you are always using the lower case versions of the constants. Doesn't it make sense to have the constants already in lower case?